iceoryx_dds/include/iceoryx_dds/gateway/iox_to_dds.hpp

```diff
@@ -26,22 +27,33 @@ namespace iox
 {
 namespace dds
 {
-///
 /// @brief DDS Gateway implementation for the iceoryx to DDS direction.
-///
 template <typename channel_t = gw::Channel<popo::UntypedSubscriber, dds::data_writer_t>,
           typename gateway_t = gw::GatewayGeneric<channel_t>>
 class Iceoryx2DDSGateway : public gateway_t
 {
   public:
     Iceoryx2DDSGateway() noexcept;
+
+    /// @brief Load the provided gateway configuration; a channel is setup if required (no error check on channel setup)
+    /// @param[in] config The gateway configuration to load
     void loadConfiguration(const config::GatewayConfig& config) noexcept;
+
+    /// @brief Discover messages coming from iceoryx.
+    /// @param[in] msg The discovery message.
     void discover(const capro::CaproMessage& msg) noexcept;
+
+    /// @brief Forward data between the two terminals of the channel used by the implementation.
+    /// @param[in] channel The channel to propagate data across.
     void forward(const channel_t& channel) noexcept;
 
   private:
+    /// @brief Setup the channel for the given service
+    /// @param[in] service service description of the service to create a channel for
+    /// @param[in] subscriberOptions the Subscriber Options options with historyCapacity and queueCapacity
+    /// @return an expected containing a copy of the added channel, otherwise an error
```

Comment:
this is also private and should not be documented with doxygen

Reply:
done